### ISSUE TITLE
Add support for $ref keyword

### ISSUE THREAD

**Issue by s1anden** [remoteoss/json-schema-form#19]
Hello! 

First off I just want to share that this looks really great, and I'd love to be able to make use of the package! 

Currently however I'm blocked by the missing support for the `$ref` keyword - I'm working with a pretty complicated schema (it's defined in multiple different files, but is almost 1000 lines when bundled together), and using internal `$ref`s for organizational purposes.

I think there's a workaround here that would work for _most_ of my use case, which is just to dereference those `$ref`s in all locations when rendering the form; however we'd really prefer not to have to do that if at all possible!

Would you all be open to adding this to your roadmap, and if so, what sort of timeline do you think we'd be looking at for getting that change in?

**Comment by sandrina-p** [comment]
Hi Samantha! @s1anden

It's true, JSF does not support `$ref` keyword as we at Remote didn't need it. The `$ref` feature can be highly complex depending on which references you are talking about. 

Are you talking about [local (`$defs`)](https://json-schema.org/understanding-json-schema/structuring.html#defs) or [Base URI (`$ref`)](https://json-schema.org/understanding-json-schema/structuring.html#id18). Would you need recursion too? 

For example, implementing Base URI seems highly complex as it would require async network request. But local `$defs` seems more easily to achieve.

May I ask you for a few JSON Schema demos/examples to evaluate exactly which feature you need?

Regarding roadmap, we don't have this in our roadmap, but if it's easy, we might add it sometime this year.

**Comment by sandrina-p** [comment]
Closing in favor of https://github.com/remoteoss/json-schema-form/issues/76. Please, follow that issue for updates :)
